spec/models/assets/import_spec.rb

```diff
@@ -451,23 +451,16 @@
       )
     end
     let(:full_remote_labware) do
-      build_remote_tube(
-        barcode: generate(:barcode),
-        uuid: remote_labware.uuid,
-        labware_barcode: remote_labware.labware_barcode
-      )
+      build_remote_tube(barcode: generate(:barcode), uuid: remote_labware.uuid, labware_barcode: remote_barcode)
     end
     let(:local_asset) { Asset.create!(barcode: local_barcode) }
 
     context 'a tube and an unknown barcode' do
       before do
         local_asset
-        expect(SequencescapeClient).to receive(:labware)
-          .with(barcode: [remote_barcode, non_existant_barcode])
-          .and_return([remote_labware])
-
-        # We still need this as we're currently immediately refreshing the resource from SS
-        allow(SequencescapeClient).to receive(:find_by_uuid).with(remote_labware.uuid).and_return(full_remote_labware)
+        stub_request(:get, %r{api/v2/labware}).to_return(
+          File.new('./spec/support/responses/sequencescape/v2/labware_tube_response.txt')
+        )
```

Comment:
While stepping away from the purity of unit tests, I've found stubbing the requests, rather than methods, is much less prone to error.